Theme showcase: Update theme support forums link

KokkieH opened the pull request:
#### Proposed Changes

* Updates the "Visit forum" button for the theme forum card on the theme support page to link to the main WordPress.com support forums, instead of the themes subforum.
* Updates unit tests for `getThemeForumUrl` to test for the new URL
* Passes the URL through `localizeUrl()` so that it redirects to the non-English forums if we have one for the user's locale

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Switch to this PR, navigate to `/themes/[site]` on a Simple, Atomic and Jetpack-connected site in turn
* Select a WordPress.com theme (simple or Premium), from the showcase
* On the theme description page, click on the Support tab
* Click the "Visit forum" button on the card reading "Have a question about this theme?"
![Videomaker_Theme_‹_kokkiestrialtestsite_—_WordPress_com](https://user-images.githubusercontent.com/11873759/175946071-3e2921d3-8b31-49f2-be99-176664607d3c.png)
* You should get taken to the main WordPress.com support forums at https://wordpress.com/forums/
* Go back to the theme page and edit the URL to view the Support tab for a WordPress.org theme instead, e.g. `theme/storefront/support/[site]`
* Click the "Visit forum" button
* You should get taken to the theme's support forum on WordPress.org

**Testing localization**

* Visit `/me/account` and change your account language to one that has its own forum, e.g. Spanish, French, Dutch, Portuguese
* Navigate to `/themes/[site]`, select a WordPress.com theme, and click on the Support tab
* Click the "Visit forum" button
* You should get taken to the localized forum for your selected language
* Change your account settings to a language that doesn't have its own forum, e.g. Afrikaans
* View the theme support page and click the "Visit forum" button
* You should get taken to the English language forums at https://wordpress.com/forums/

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes #64976

KokkieH committed:
Update theme forum URL (#64976)

We're shutting down the subforums on WordPress.com, including the themes subforum.
This changes the theme forum URL to go to the main support forums instead
for WordPress.com themes.

Also updating the comment to accurately reflect what the code does, as it
still references premium theme forums which also no longer exist.

KokkieH committed:
Update tests for theme forum URL (#64976)

Changes the tests for getThemeForumUrl to expect the general WordPress.com
support forums, rather than the theme support subforum, as we're retiring
the theme support subforums on WordPress.com

yoavf commented on client/state/themes/selectors/get-theme-forum-url.js:
```diff
@@ -4,16 +4,16 @@ import { isWporgTheme } from 'calypso/state/themes/selectors/is-wporg-theme';
 export function getThemeForumUrl( state, themeId ) {
 	if ( isWpcomTheme( state, themeId ) ) {
-		return '//en.forums.wordpress.com/forum/themes';
+		return '//en.forums.wordpress.com/';
```
- Let's use the current canonical url instead: `wordpress.com/forums`

KokkieH replied:
Updated in https://github.com/Automattic/wp-calypso/pull/64987/commits/6ee06385c24f813aa452c620099d289f160ea964, for the tests as well

yoavf commented on client/state/themes/selectors/get-theme-forum-url.js:
```diff
@@ -4,16 +4,16 @@ import { isWporgTheme } from 'calypso/state/themes/selectors/is-wporg-theme';
 export function getThemeForumUrl( state, themeId ) {
 	if ( isWpcomTheme( state, themeId ) ) {
-		return '//en.forums.wordpress.com/forum/themes';
+		return '//en.forums.wordpress.com/';
```
@KokkieH What do you think about passing the URL through `localizeUrl()` so that it redirects to the non-English forums if we have any for the user's locale?

KokkieH replied:
That would be consistent with the help centre and contact form, so I can try to add that, yes.

KokkieH replied:
I've made this change in https://github.com/Automattic/wp-calypso/pull/64987/commits/ce72a4b723fe3dd8f26cf25ff0cc1229bcbed750

KokkieH committed:
Update forum URL (#64987)

Update forums URL, which used to be en.forums.wordpress.com
to canonical URL, wordpress.com/forums instead

KokkieH committed:
Localize theme forum URL (#64976)

We're shutting down the themes subforum on WordPress.com, updating the forums
support card to link to the forums homepage instead

This localizes the URL, so if the user has a locale set which has its own
forum, they'll get taken to that forum instead.

yoavf reviewed (approved):
Great work @KokkieH, this is ready to merge!
